src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -94,183 +95,133 @@ struct npc_corastrasza : public ScriptedAI
-    GO_CARIBOU_TRAP_9   = 188002,
-    GO_CARIBOU_TRAP_10  = 188003,
-    GO_CARIBOU_TRAP_11  = 188004,
-    GO_CARIBOU_TRAP_12  = 188005,
-    GO_CARIBOU_TRAP_13  = 188006,
-    GO_CARIBOU_TRAP_14  = 188007,
-    GO_CARIBOU_TRAP_15  = 188008,
-
-    SPELL_TRAPPED       = 46104,
-
-    // Texts
-    SAY_NESINGWARY_1    = 0
-};
-
-#define CaribouTrapsNum 15
-const uint32 CaribouTraps[CaribouTrapsNum] =
-{
-    GO_CARIBOU_TRAP_1, GO_CARIBOU_TRAP_2, GO_CARIBOU_TRAP_3, GO_CARIBOU_TRAP_4, GO_CARIBOU_TRAP_5,
-    GO_CARIBOU_TRAP_6, GO_CARIBOU_TRAP_7, GO_CARIBOU_TRAP_8, GO_CARIBOU_TRAP_9, GO_CARIBOU_TRAP_10,
-    GO_CARIBOU_TRAP_11, GO_CARIBOU_TRAP_12, GO_CARIBOU_TRAP_13, GO_CARIBOU_TRAP_14, GO_CARIBOU_TRAP_15,
-};
+/*######
+## Quest 11865: Unfit for Death
+######*/
 
-// 46085 - Place Fake Fur
-class spell_q11865_place_fake_fur : public SpellScript
+// Gameobjects 187982,187995,187996,187997,187998,187999,188000,188001,188002,188003,188004,188005,188006,188007,188008: Caribou Trap
+enum CaribouTrap
 {
-    PrepareSpellScript(spell_q11865_place_fake_fur);
-
-    bool Load() override
-    {
-        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
-    }
-
-    void ActivateGameObject(SpellEffIndex effIndex)
-    {
-        PreventHitDefaultEffect(effIndex);
-        GameObject* go = GetHitGObj();
-        Player* player = GetCaster()->ToPlayer();
-
-        if (go->FindNearestCreature(NPC_NESINGWARY_TRAPPER, 10.0f, true) || go->FindNearestCreature(NPC_NESINGWARY_TRAPPER, 10.0f, false) || go->FindNearestGameObject(GO_HIGH_QUALITY_FUR, 2.0f))
-            return;
-
-        float x, y, z;
-        go->GetClosePoint(x, y, z, go->GetCombatReach() / 3, 7.0f);
-
-        go->SummonGameObject(GO_HIGH_QUALITY_FUR, go->GetPosition(), QuaternionData(), 20s);
-        if (TempSummon* summon = player->SummonCreature(NPC_NESINGWARY_TRAPPER, x, y, z, go->GetOrientation(), TEMPSUMMON_DEAD_DESPAWN, 1s))
-        {
-            summon->SetVisible(false);
-            summon->SetReactState(REACT_PASSIVE);
-            summon->SetImmuneToPC(true);
-        }
-    }
-
-    void Register() override
-    {
-        OnEffectHitTarget += SpellEffectFn(spell_q11865_place_fake_fur::ActivateGameObject, EFFECT_0, SPELL_EFFECT_ACTIVATE_OBJECT);
-    }
+    EVENT_FUR_SPAWN        = 1,
+    EVENT_SPAWN_TRAPPER    = 2,
+    EVENT_TRAPPER_MOVE     = 3,
+    EVENT_TRAPPER_TEXT     = 4,
+    EVENT_TRAPPER_LOOT     = 5,
+    EVENT_FUR_DESPAWN      = 6,
+    EVENT_TRAPPER_DIE      = 7,
+    EVENT_DESPAWN_ALL      = 8,
+    GO_HIGH_QUALITY_FUR    = 187983,
+    NPC_NESINGWARY_TRAPPER = 25835,
+    SAY_NESINGWARY_1       = 0,
+    SPELL_PLACE_FAKE_FUR   = 46085,
+    SPELL_TRAPPED          = 46104
 };
 
-struct npc_nesingwary_trapper : public ScriptedAI
+struct go_caribou_trap : public GameObjectAI
 {
-    npc_nesingwary_trapper(Creature* creature) : ScriptedAI(creature)
+    go_caribou_trap(GameObject* go) : GameObjectAI(go)
     {
-        Initialize();
+        InitializeAI();
```

Comment:
Please only initialize class member fields in the constructor. Every AI logic and action should be placed somewhere else, either Reset() or whatever the base AI class makes available.